hibernate-core/src/test/java/org/hibernate/orm/test/util/StandardStackTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.orm.test.util;
+
+import java.util.NoSuchElementException;
+
+import org.hibernate.internal.util.collections.Stack;
+import org.hibernate.internal.util.collections.StandardStack;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * @author Marco Belladelli
+ */
+public class StandardStackTest {
+	@Test
+	public void testSimpleStackAccess() {
+		final Stack<Integer> stack = allocateStack( 5 );
+		assertEquals( 5, stack.depth() );
+		assertEquals( 0, stack.getRoot() );
+		assertEquals( 4, stack.getCurrent() );
+		assertEquals( 4, stack.pop() );
+		assertEquals( 4, stack.depth() );
+	}
+
+	@Test
+	public void testClear() {
+		final Stack<Integer> stack = allocateStack( 42 );
+		assertEquals( 42, stack.depth() );
+		assertFalse( stack.isEmpty() );
+		stack.clear();
+		assertEquals( 0, stack.depth() );
+		assertTrue( stack.isEmpty() );
```

Comment:
let's also test for that the various "find" methods work correctly on both an empty and a cleared stack?